specification.md

```diff
@@ -1574,3 +1579,119 @@ to send over the connection. When using a single connection to transmit
 larger messages, TCP buffers can be more efficiently utilized.
 Additionally, it becomes feasible to tune buffer sizes in order to
 optimize the throughput of connections to particular services.
+
+
+
+### Appendix C - examples
+
+This Appendix contains some examples of how the API can be used. Unlike the
+specification language, the examples are in a hypothetical Python binding, which
+is expected to be a relatively frequently used binding.
+
+#### Submit and wait for N jobs
+
+This example shows how to submit `N` jobs and synchronously wait for them to
+complete.
+
+```python
```

Comment:
Would you mind if I add a short note that the Python code used in examples is not normative and that Python bindings may deviate? :-)

Reply:
I don't see where the degree of freedom is. The API should be constraining enough such that such notes should not be necessary. I mean I'm worried about sending the wrong message here, which is that the spec doesn't really matter: implementations should briefly read it and do whatever they want anyway.

Reply:
Well, we wanted to eventually have a separate binding specification (see #27), or at least that is how I remember our last discussion on this topic went?  Well, what I am really getting at is this: I would do things different in Python, but I don't want to muddy the waters on this PR by discussing how the Python code should look like, when the point is to add examples which demonstrate what functionality / semantics the API tries to cover.

Reply:
I think my initial reply wasn't quite useful. Perhaps it would be more useful to see what you had in mind, and maybe just use that style for the examples.